addons/web_editor/static/src/js/editor/odoo-editor/src/utils/sanitize.js

```diff
@@ -79,186 +73,125 @@ export function areSimilarElements(node, node2) {
-                        !node.parentElement.classList.contains('o_checklist') ||
-                        [...node.children].some(child => ['UL', 'OL'].includes(child.nodeName))
-                )) {
-                    // Remove unique ids from checklists and stars from elements
-                    // that are no longer checklist items or stars, and from
-                    // parents of nested lists.
-                    node.removeAttribute('id')
-                } else {
-                    // Add/change IDs where needed, and ensure they're unique.
-                    let id = node.getAttribute('id');
-                    if (!id || ids.includes(id)) {
-                        id = `checkId-${nextId}`;
-                        nextId++;
-                        node.setAttribute('id', id);
-                    }
-                    ids.push(id);
-                }
-            }
-        }
+/**
+ * Sanitize the given node and return it.
+ *
+ * @param {Node} node
+ * @param {Element} root
+ * @returns {Node} the sanitized node
+ */
+function sanitizeNode(node, root) {
+    // First ensure elements which should not contain any content are tagged
+    // contenteditable=false to avoid any hiccup.
+    if (isVoidElement(node) && node.getAttribute('contenteditable') !== 'false') {
+        node.setAttribute('contenteditable', 'false');
     }
 
-    parse(node) {
-        node = closestBlock(node);
-        if (node && ['UL', 'OL'].includes(node.tagName)) {
-            node = node.parentElement;
+    if (
+        areSimilarElements(node, node.previousSibling) &&
+        !isUnbreakable(node) &&
+        !isEditorTab(node) &&
+        !(
+            node.attributes?.length === 1 &&
+            node.hasAttribute('data-oe-zws-empty-inline') &&
+            (node.textContent === '\u200B' || node.previousSibling.textContent === '\u200B'))
+    ) {
+        // Merge identical elements together.
+        getDeepRange(root, { select: true });
+        const restoreCursor = node.isConnected && preserveCursor(root.ownerDocument);
+        moveNodes(...startPos(node), node.previousSibling);
+        restoreCursor?.();
+    } else if (
+        node.nodeName === 'P' && // Note: not sure we should limit to <p>.
+        node.parentElement.nodeName === 'LI' &&
+        isEmptyBlock(node)
+    ) {
+        // Remove empty paragraphs in <li>.
+        const parent = node.parentElement;
+        const restoreCursor = node.isConnected && preserveCursor(root.ownerDocument);
+        node.remove();
+        fillEmpty(parent);
+        restoreCursor?.(new Map([[node, parent]]));
+        node = parent;
+    } else if (node.nodeName === 'LI' && !node.closest('ul, ol')) {
+        // Transform <li> into <p> if they are not in a <ul> / <ol>.
+        const paragraph = document.createElement('p');
+        paragraph.replaceChildren(...node.childNodes);
+        node.replaceWith(paragraph);
+        node = paragraph;
```

Comment:
Same here.

Reply:
`// The node has been removed, we need to update it.` ?

Reply:
Same as above, obviously.